hydrolib/core/io/ini/parser.py

```diff
@@ -239,8 +239,26 @@ def _handle_datarow(self, line: str) -> None:
 
     def _retrieve_property_comment(self, line: str) -> Tuple[Optional[str], str]:
         if self._config.parse_comments and self._config.comment_delimiter in line:
-            key_value, comment = line.split(self._config.comment_delimiter, 1)
-            return comment.strip(), key_value.strip()
+            # key_value, comment = line.split(self._config.comment_delimiter, 1)
+            parts = line.strip().split(self._config.comment_delimiter)
+            numhash = len(parts) - 1
```

Comment:
Might be better to just use `line.count("#")`, and only split/strip when really required.

Reply:
is already done by existing if in line above: `[..]and self._config.comment_delimiter in line:`